### ISSUE TITLE
Cannot compile with info.picocli:picocli-spring-boot-starter dependency

### ISSUE THREAD

**Issue by remal** [remkop/picocli#1294]
I'm using `info.picocli:picocli-spring-boot-starter:4.6.0` as a Gradle dependency.

Looks like that after [fixing Gradle warnings](https://github.com/remkop/picocli/commit/758558b9817660b6c6e497aab28cb8ac7421127e) `info.picocli:picocli:4.6.0` transitive dependency became runtime and its classes can't be used at compile-time.

https://github.com/remkop/picocli/blob/572e05f98a39af074cc4872ed1eb4261fad311c2/picocli-spring-boot-starter/build.gradle#L19-L20
I suppose `api` should be used instead of `implementation`, as Gradle treats `implementation` as runtime only dependencies.

Take a look: https://mvnrepository.com/artifact/info.picocli/picocli-spring-boot-starter/4.6.0
![Maven Repository - info picocli](https://user-images.githubusercontent.com/231188/103465935-59402d80-4d51-11eb-81b1-cd5db498c0f6.png)

**Comment by remal** [comment]
I suggest changing this: https://github.com/remkop/picocli/blob/572e05f98a39af074cc4872ed1eb4261fad311c2/picocli-spring-boot-starter/build.gradle#L19-L20

... to:
```
 api     rootProject 
 api     "org.springframework.boot:spring-boot-starter:$springBootVersion" 
```

This change should fix the issue.

**Comment by remkop** [comment]
Thank you for raising this!
This is fixed in the 4.6.1 release.
